core/crypto.mk

```diff
@@ -82,17 +82,18 @@ $(call force,CFG_AES_GCM_TABLE_BASED,n,conflicts with CFG_CRYPTO_WITH_CE)
 CFG_HWSUPP_PMULT_64 ?= y
 
 ifeq ($(CFG_ARM32_core),y)
-CFG_CRYPTO_AES_ARM32_CE ?= $(CFG_CRYPTO_AES)
 CFG_CRYPTO_SHA1_ARM32_CE ?= $(CFG_CRYPTO_SHA1)
 CFG_CRYPTO_SHA256_ARM32_CE ?= $(CFG_CRYPTO_SHA256)
 endif
 
 ifeq ($(CFG_ARM64_core),y)
-CFG_CRYPTO_AES_ARM64_CE ?= $(CFG_CRYPTO_AES)
 CFG_CRYPTO_SHA1_ARM64_CE ?= $(CFG_CRYPTO_SHA1)
 CFG_CRYPTO_SHA256_ARM64_CE ?= $(CFG_CRYPTO_SHA256)
 endif
 
+CFG_CRYPTO_AES_ARM_CE ?= $(CFG_CRYPTO_AES)
+CFG_CRYPTO_AES_ACCEL ?= $(CFG_CRYPTO_AES_ARM_CE)
```

Comment:
What's the difference between `CFG_CRYPTO_AES_ACCEL` and `CFG_CRYPTO_AES_ARM_CE` (is it really needed)? Assuming there is one, should `_ACCEL` be exposed to the user or rather be internal (`_CFG`)?

Reply:
I'm trying to have two levels of abstraction. `_ACCEL` is for code outside `core/arch/arm` while `_ARM_CE` is specific for Arm.

More ways of accelerating these may become available later on, perhaps CAAM.

This shouldn't concern user mode. Perhaps we should have a `CORE_CRYPTO_` prefix instead of just a `CRYPTO_` prefix. What do you say?

Reply:
I see, makes sense.
Yeah `CFG_CORE_CRYPTO_*_ACCEL` sounds good to me. And maybe add a not  mentioning that `CFG_CRYPTO_*_ARM_CE` applies to both core and user space? I had completely missed that!

Reply:
It turns out I was wrong in that. The `CFG_CRYPTO_*_ARM_CE` are not exported to user space.

Reply:
OK fine